packages/core/src/asyncContext/stackStrategy.ts

```diff
@@ -131,20 +132,17 @@ export class AsyncContextStack {
  * This will be removed during the v8 cycle and is only here to make migration easier.
  */
 function getAsyncContextStack(): AsyncContextStack {
-  const registry = getMainCarrier();
-
-  // For now we continue to keep this as `hub` on the ACS,
-  // as e.g. the Loader Script relies on this.
-  // Eventually we may change this if/when we update the loader to not require this field anymore
-  // Related, we also write to `hub` in {@link ./../sdk.ts registerClientOnGlobalHub}
-  const sentry = getSentryCarrier(registry) as { hub?: AsyncContextStack };
-
-  if (sentry.hub) {
-    return sentry.hub;
+  // eslint-disable-next-line deprecation/deprecation
+  const sentry = getSentryCarrier(getMainCarrier()) as { hub?: AsyncContextStack };
+  if (!sentry.hub) {
+    // For now, we still set the `hub` object to gracefully handle pre v8 SDKs and certain scripts
+    // (e.g. our loader script) accessing properties on `window.__SENTRY__.hub`.
+    // This will just call the `getCurrentHubShim` function, as if users would interact with `getCurrentHub`.
+    // eslint-disable-next-line deprecation/deprecation
+    Object.defineProperty(sentry, 'hub', { get: () => getCurrentHubShim(), enumerable: true });
```

Comment:
Second oof: Referencing `getCurrentHubShim` here actually increases bundle size by +1% because previously, `getCurrentHub(Shim)` was completely optional :(